src/components/SortAndSearch.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+	Autocomplete,
+	Button,
+	FormControl,
+	InputLabel,
+	MenuItem,
+	Select,
+	SelectChangeEvent,
+	TextField,
+} from '@mui/material'
+import { ProductCardProps } from './ProductCard'
+import { Dispatch, SetStateAction, SyntheticEvent, useCallback, useEffect, useState } from 'react'
+import { exampleProducts } from '~/data/exampleProducts'
+
+interface SortAndSearchProps {
+	filteredProducts: ProductCardProps[]
+	setProducts: Dispatch<SetStateAction<ProductCardProps[]>>
+	setSearchTerm: Dispatch<SetStateAction<string>>
+	toggleDrawer: (newOpen: boolean) => () => void
+}
+
+const SortAndSearch = ({ setProducts, toggleDrawer, filteredProducts, setSearchTerm }: SortAndSearchProps) => {
+	const [sortOption, setSortOption] = useState('popularity-desc')
+
+	useEffect(() => {
```

Comment:
Couldn't this effect be a separate method, which will be triggered by some actions by the user?